ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/services/RestServiceContextFactory.java

```diff
@@ -74,6 +74,7 @@ public static ServiceContext create(
         srClientFactory,
         () -> new DefaultConnectClient(ksqlConfig.getString(KsqlConfig.CONNECT_URL_PROPERTY),
                                        authHeader),
+        // We should be re-using this not creating one each time!
```

Comment:
The client suppliers are memoized: https://github.com/confluentinc/ksql/blob/9bc22207a57947beb1b922c1e923b7635d75e903/ksql-engine/src/main/java/io/confluent/ksql/services/DefaultServiceContext.java#L96
Does that address the concern?

Reply:
Ah I see, so we are reusing :) In which case I can retain the larger pool size.